### PR TITLE
Function defaults

### PR DESCRIPTION
**Pull request by revmischa**
Add the ability to specify to the Generator construct some lambda defaults for all functions

This is nice because you can define some common attributions for your functions or the compiler options

### CODE REVIEW

File: src/app/config.ts [hunk 1]
```diff
@@ -0,0 +1,14 @@
...
+ *
+ * @module
+ */
+
+export { CorsHttpMethod, CorsPreflightOptions } from "@aws-cdk/aws-apigatewayv2"
+
+// currently unused
+export interface IConfigGeneric {}
+
+export type IConfig<T = Record<string | number | symbol, unknown>> = IConfigGeneric & T
```
Comment (revmischa) [inline src/app/config.ts:14]: Just a placeholder for now, not sure if this is needed or not

File: README.md [hunk 2]
```diff
@@ -1,9 +1,9 @@
 # JetKit/CDK
 
-An opinionated toolkit for building cloud-native serverless applications.
+An anti-framework for building cloud-native serverless applications.
 
-This module provides tools for defining RESTful API views with code
-and generating cloud infrastructure using [AWS CDK](https://docs.aws.amazon.com/cdk/latest/guide/home.html).
+This module provides convenient tools for write RESTful API views
```
Comment (Dizzzmas) [inline README.md:5]:
```suggestion
This module provides convenient tools for writing RESTful API views
```

File: src/cdk/generator.ts [hunk 3]
```diff
@@ -78,9 +87,13 @@ export class ResourceGenerator extends Construct {
     let apiViewConstruct: undefined | ApiViewConstruct
     if (apiViewMeta) {
       const name = apiViewMeta.apiClass.name
+
+      // merge defaults with config in meta
+      const mergedOptions: Omit<ApiProps, "httpApi"> = deepmerge(this.functionOptions || {}, apiViewMeta)
```
Comment (Dizzzmas) [inline src/cdk/generator.ts:92]: Sorry, I don't understand what this achieves and why

File: src/cdk/lambda/node14func.ts [hunk 4]
```diff
@@ -32,17 +34,19 @@ export class Node14Func extends NodejsFunction {
...
       bundling: {
         keepNames,
-        bundlingRest,
+        ...bundlingRest,
       },
     }
 
     super(scope, id, newProps)
+
+    this.bundling = bundling
```
Comment (Dizzzmas) [inline src/cdk/lambda/node14func.ts:50]: Are we overriding here the default cdk construct for Lambdas? What does bundling do here?

File: src/cdk/generator.ts [hunk 5]
```diff
@@ -25,6 +27,11 @@ export interface ResourceGeneratorProps {
    * The {@link HttpApi} to attach routes to.
    */
   httpApi: HttpApi
+
+  /**
+   * Default Lambda function options.
+   */
+  functionOptions?: NodejsFunctionProps
```
Comment (Dizzzmas) [inline src/cdk/generator.ts:34]: Will we be able to get the idea of what the usual options are by looking at the interface or it's generic and we need some examples of what can be passed there?

**Review by Dizzzmas — APPROVED** [review]

File: src/cdk/lambda/node14func.ts (hunk 4 above)
Comment (revmischa) [reply src/cdk/lambda/node14func.ts:50]: It's just informational, I only added it to verify it in the test. It doesn't have any function

File: src/cdk/generator.ts (hunk 5 above)
Comment (revmischa) [reply src/cdk/generator.ts:34]: You can see them here https://jetbridge.github.io/jetkit-cdk/interfaces/iapiviewclassmetadata.html

File: src/cdk/generator.ts (hunk 3 above)
Comment (revmischa) [reply src/cdk/generator.ts]: I updated the comment hopefully it's clearer

**Review by rkyslyy — APPROVED** [review]

**Pull request merged**